src/main/java/duke/common/InputChecker.java

```diff
@@ -0,0 +1,43 @@
+package duke.common;
+
+import duke.DukeException;
+import duke.ui.BotUI;
+
+/**
+ * Responsible in the initial checks of Adding tasks' input
+ * is in the correct format.
+ *
+ */
+public class InputChecker {
+
+    private static final BotUI UI = new BotUI();
+
+    private static void checkDeadlineAndEvent(String input, String phrase, String taskSplitter) throws DukeException {
+        String description = input.replace(phrase, "").replace(" ", "");
+        if (!description.contains(taskSplitter) || description.substring(0, description.indexOf(taskSplitter)).isEmpty()

```

Comment:
Minor nitpick, but you can improve the readability of this by defining self-explanatory booleans to store your conditionals